crates/bevy_asset/src/server/mod.rs

```diff
@@ -270,7 +270,29 @@ impl AssetServer {
     /// The asset load will fail and an error will be printed to the logs if the asset stored at `path` is not of type `A`.
     #[must_use = "not using the returned strong handle may result in the unexpected release of the asset"]
     pub fn load<'a, A: Asset>(&self, path: impl Into<AssetPath<'a>>) -> Handle<A> {
-        self.load_with_meta_transform(path, None)
+        self.load_with_meta_transform(path, None, ())
+    }
+
+    /// Begins loading an [`Asset`] of type `A` stored at `path` while holding a guard item.
+    /// The guard item is dropped when either the asset is loaded or loading has failed.
+    ///
+    /// This function returns a "strong" [`Handle`]. When the [`Asset`] is loaded (and enters [`LoadState::Loaded`]), it will be added to the
+    /// associated [`Assets`] resource.
+    ///
+    /// The guard item should notify the caller in its [`Drop`] implementation. See example `multi_asset_sync`.
+    /// Synchronously this can be a [`Arc<AtomicU32>`] that decrements its counter, asynchronously this can be a `Barrier`.
+    ///
+    /// Additionally, you can check the asset's load state by reading [`AssetEvent`] events, calling [`AssetServer::load_state`], or checking
+    /// the [`Assets`] storage to see if the [`Asset`] exists yet.
+    ///
+    /// The asset load will fail and an error will be printed to the logs if the asset stored at `path` is not of type `A`.
+    #[must_use = "not using the returned strong handle may result in the unexpected release of the asset"]
+    pub fn load_acquire<'a, A: Asset, G: Send + Sync + 'static>(
```

Comment:
I didn't expect the builder-pattern to land quite so quickly (sorry about that), but it might be worth replacing these two functions with one in `NestedLoader`.

It might also make sense to support it for untyped handles, but I don't think that has to happen now.

Reply:
Yeah I think getting these ready is more important to keep the scope of this PR small. 
There is already a public `async` function for loading untyped assets so half of the use cases are already covered.